backend/app/utils/data_fetchers.py

```
@@ -29,88 +29,51 @@ def load_data(self, symbol: str, source: str = None, days: int = 365) -> pd.Data
-        common_mappings = {
-            'btc': 'bitcoin',
-            'eth': 'ethereum',
-            'bnb': 'binancecoin',
-            'xrp': 'ripple',
-            'ada': 'cardano',
-            'sol': 'solana',
-            'doge': 'dogecoin'
+        self.api_key = api_key
+        self.base_url = "https://pro-api.coinmarketcap.com"
+
+    def load_data(self, symbol: str, source: str = "coinmarketcap", days: int = 365) -> pd.DataFrame:
```

Pylint (3.3.7)

[refactor] Too many local variables (16/15)

(R0914)